JustLog/Classes/LogstashDestination.swift

```diff
@@ -9,66 +9,71 @@
 import Foundation
 import SwiftyBeaver
 
-/// This entire class relies on `logDispatchQueue` to synchronise access to the `logsToShip` dictionary
-/// Every action is put in the `logDispatchQueue` which is synchronous, even though the actual sending
-/// via `URLSession` is not.
+typealias LogContent = [String: Any]
+typealias LogTag = Int
+
+/// This entire class relies on `operationQueue` to synchronise access to the `logsToShip` dictionary
+/// Every action is put in the `operationQueue` which is synchronous, even though the actual sending
+/// via `URLSession` is not. The writers' completion handler are executed on the underlying queue of the `OperationQueue`
 /// Here's a brief summary of what operations are added to the queue and when:
 /// A log is generated via `send` and added to the `logsToShip` dictionary (along with a `tag` identifier).
 /// At some point `writeLogs` is called that creates an `NSURLSessionStreamTask` to send the log to the server.
-/// When the task completes, either `completeLog` or `retryLog` are called.
-///    `completeLog` removes the log from `logsToShip`
-///    `retryLog` clears the log from `logsInSocketQueue` and causes it to be resent the next time `writeLogs` is called.
-/// An optional `completionHandler` is called when all logs are written.
+/// When the writer finish all streams operation it calls the completion handler passing the logs that failed to push.
```

Comment:
When the writer finishes all stream operations, it calls [...]
or
Once the writer has completed all operations, it calls [...]